lib/std/stropt.nim

```diff
@@ -0,0 +1,71 @@
+#
+#
+#           The Nim Compiler
+#        (c) Copyright 2021 Nim Contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+const whitespaces = {' ', '\t', '\v', '\r', '\l', '\f'}
+
+func stripSlice*(s: openArray[char], leading = true, trailing = true, chars: set[char] = whitespaces): Slice[int] =
+  runnableExamples:
+    doAssert stripSlice(" abc  ") == 1 .. 3
+  var
+    first = 0
+    last = high(s)
+  if leading:
+    while first <= last and s[first] in chars: inc(first)
+  if trailing:
+    while last >= first and s[last] in chars: dec(last)
+  result = first .. last
+
+func setSlice*(s: var string, slice: Slice[int]) =
+  ## Inplace version for `substr`.
```

Comment:
* add a `runnableExamples`
* specify behavior for edge case where slice is out of bounds; also we need to think about what's the sanest behavior here since this is a new proc (refs https://github.com/nim-lang/Nim/issues/16492)

Reply:
I'm not sure which is better
```nim
  let first = slice.a
  let last = slice.b

  assert first >= 0
  assert last <= s.high

```
or
```nim
  let first = slice.a
  let last = slice.b

  let s = max(first, 0)
  let l = min(0, s.high)

```

Reply:
since focus of strbasics is on performance, it seems clear that `assert` version is preferred over `max, min`; client code could do that max/min on their end; so we avoid double penalty in with `--asertions:off`

Reply:
I agree